worksheet.rb

```diff
@@ -1,17 +1,197 @@
+trips = {
+    "DR0004": [
+        {
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider_id: "RD0022",
+            rating: 5
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider_id: "RD0022",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 20 ,
+            rider_id: "RD0073",
+            rating: 5
+        }
+    ],
+    "DR0001": [
+        {
+            date: "3rd Feb 2016",
+            cost: 10,
+            rider_id: "RD0003",
+            rating: 3
+        },
+        {
+            date: "3rd Feb 2016",
+            cost: 30,
+            rider_id: "RD0015",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 45,
+            rider_id: "RD0003",
+            rating: 2
+        }
+    ],
+    "DR0002": [
+        {
+            date: "3rd Feb 2016	",
+            cost: 25,
+            rider_id: "RD0073",
+            rating: 5,
+        },
+        {
+            date: "4th Feb 2016	",
+            cost: 15,
+            rider_id: "RD0013",
+            rating: 1,
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 35,
+            rider_id: "RD0066",
+            rating: 3,
+        }
+    ],
+    "DR0003": [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider_id: "RD0066",
+            rating: 5,
+        },
+        {
+            date: "5th Feb 2016	",
+            cost: 50,
+            rider_id: "RD0003",
+            rating: 2,
+        }
+    ]
+}
+# METHODS
+#method to figure out number of rides each driver has given
+def num_of_rides(trips)
+  return trips.map do |driver, trip|
+    "Driver #{driver} made #{trip.length} rides"
+  end
+end
+
+# method for total amount of money each driver has made
+def total_earned(trips)
+  result = trips.map do |driver, trip|
+    amount = trip.map { |ride| ride[:cost]}.sum
+    { :driver => driver.to_s, :amount => amount}
+  end
+  return result
+end
+
+# method for average rating for each driver
+def average_rating(trips)
+  result = trips.map do |driver, trip|
+    amount = trip.map { |ride| ride[:rating]}.sum
+    average_rating = (amount.to_f/trip.size).round(2)
+    {:driver => driver, :avg_rating => average_rating}
+  end
+  return result
+end
+
+# method to find Which Driver made the most money
+def driver_with_cash(total_earned)
+  return total_earned.max_by { |driver_earning| driver_earning[:amount]}
+end
+
+# method to find which driver had the highest rating
+def driver_with_highest_rating(average_rating)
+  return average_rating.max_by { |driver_rating| driver_rating[:avg_rating]}
+end
+
+#CALLING my methods below:
```

Comment:
Really great decision to organize your code this way: methods on top and driver code below!